File: 3043-find-the-length-of-the-longest-common-prefix/3043-find-the-length-of-the-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, arr1: List[int], arr2: List[int]) -> int:
        ans = 0
        prefixes = set()

        for v in arr1:
            while v > 0 and v not in prefixes:
                prefixes.add(v)
                v //= 10
        
        for v in arr2:
            while v > 0 and v not in prefixes:
                v //= 10
            
            if v == 0: continue

            ans = max(ans, len(str(v)))

        return ans

# array, hash map
# time: O(nlog(d) + mlog(d))
# space: O(nlog(d))

# #1 2026/05/21

File: 3043-find-the-length-of-the-longest-common-prefix/3043-find-the-length-of-the-longest-common-prefix.go
func longestCommonPrefix(arr1 []int, arr2 []int) int {
	ans := 0
	m1 := map[int]bool{}

	for _, a1 := range arr1 {
		for a1 > 0 && !m1[a1] {
			m1[a1] = true
			a1 /= 10
		}
	}

	for _, a2 := range arr2 {
		for a2 > 0 && !m1[a2] {
			a2 /= 10
		}

		if a2 == 0 {
			continue
		}

		ans = max(ans, len(strconv.Itoa(a2)))
	}

	return ans
}

// array, hash map

// #1 2026/05/21